src/command.rs

```diff
@@ -139,3 +167,89 @@ impl ShellCommand for ExternalCommand {
         Ok(child.stdout.take())
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct LlmCommand {
+    prompt: String,
+    openai_client: OpenAIClient,
+}
+
+impl LlmCommand {
+    pub fn new(prompt: String, openai_client: OpenAIClient) -> Self {
+        LlmCommand {
+            prompt,
+            openai_client,
+        }
+    }
+
+    pub async fn generate_response(&self, input: Option<String>) -> Result<String, Box<dyn Error>> {
+        let context = if let Some(input) = input {
+            format!("{}: {}", self.prompt, input)
+        } else {
+            self.prompt.clone()
+        };
+        let output = self.openai_client.generate_text(&context, 100).await?;
+        Ok(output)
+    }
```

Comment:
We probably want to have some extra prompting to help the LLM generate good responses.
1. Adding a system prompt
2. Structuring the prompt to distinguish between the prompt and the inputs (from another command)
3. Probably best not to limit the output by 100? Or at least provide a way to configure it / turn output limiting off in #6.

Reply:
We may want to explore more advanced prompting techniques in the future, but this should do in the mean time.

Reply:
Yeah I agree, more advanced prompting would be good in the future